### PR TITLE
Remove size check between attn_weights and kv_seq_len for phi3

### PR DESCRIPTION
**Pull request by helunwencser**
# What does this PR do?

As shown in this [issue](https://github.com/huggingface/transformers/issues/32338), Phi3 current fails to work with StaticCache because StaticCache does not slice `k_out, v_out` upon `update` while Phi3 [expects](https://github.com/huggingface/transformers/blob/main/src/transformers/models/phi3/modeling_phi3.py#L404-L408) it to only return cache for previous seen tokens.

In the long term, https://github.com/huggingface/transformers/pull/31421 and https://github.com/huggingface/transformers/pull/30862 should resolve this problem by supporting StaticCache and dynamic length.

For now, removing this size check should make Phi3 work with StaticCache.

Fixes # (issue)
https://github.com/huggingface/transformers/issues/32338

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@ArthurZucker , @zucchini-nlp

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
It's not even there for the sdpa, let's go with it. If you can add a super small tests showing how you use this with sstatic cache would be lovely!

**Comment by helunwencser** [comment]
Thanks! Added integration tests for each one. Also fixed the existing two integration tests which are failing.

**Comment by guangy10** [comment]
cc: @amyeroberts @gante

**Review by zucchini-nlp — APPROVED** [review]
Yes, we'll check out that all models support new cache types after the mentioned PR is merged. This works for Phi3, thanks for fixing!

**Comment by helunwencser** [comment]
@ArthurZucker , @zucchini-nlp thanks for reviewing this! Could you please help merge this PR as well? Looks like I cannot do that due to not having write access.

**Comment by ArthurZucker** [comment]
Thanks for the fix! Will be part of the release!

**Pull request merged**
